### ISSUE TITLE
ansible: ansible-galaxy install problem with pyopenssl

### ISSUE THREAD

**Issue by davidak** [Homebrew/homebrew-core#4416]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports:

```
$ ansible-galaxy install kbrebanov.resolv_conf -vvv
[WARNING]: log file at /var/log/ansible.log is not writeable and we cannot create it, aborting

Using /Users/davidak/code/ansible/ansible.cfg as config file
Opened /Users/davidak/.ansible_galaxy
Installing role kbrebanov.resolv_conf
Opened /Users/davidak/.ansible_galaxy
- downloading role 'resolv_conf', owned by kbrebanov
https://galaxy.ansible.com/api/v1/roles/?owner__username=kbrebanov&name=resolv_conf
ERROR! Unexpected Exception: name must be a byte string
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/bin/ansible-galaxy", line 92, in <module>
    exit_code = cli.run()
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/cli/galaxy.py", line 133, in run
    self.execute()
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/cli/__init__.py", line 103, in execute
    fn()
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/cli/galaxy.py", line 385, in execute_install
    installed = role.install()
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/galaxy/role.py", line 204, in install
    role_data = api.lookup_role_by_name(self.src)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/galaxy/api.py", line 56, in wrapped
    return method(self, *args, **kwargs)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/galaxy/api.py", line 178, in lookup_role_by_name
    data = self.__call_galaxy(url)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/galaxy/api.py", line 56, in wrapped
    return method(self, *args, **kwargs)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/galaxy/api.py", line 92, in __call_galaxy
    resp = open_url(url, data=args, validate_certs=self._validate_certs, headers=headers, method=method)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 839, in open_url
    r = urllib2.urlopen(*urlopen_args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 402, in open
    req = meth(req)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 665, in http_request
    ssl_s = ssl_wrap_socket(s, ca_certs=tmp_ca_cert_path, cert_reqs=ssl.CERT_REQUIRED, ssl_version=PROTOCOL, server_hostname=self.hostname)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/vendor/lib/python2.7/site-packages/requests/packages/urllib3/contrib/pyopenssl.py", line 296, in ssl_wrap_socket
    cnx.set_tlsext_host_name(server_hostname)
  File "/usr/local/Cellar/ansible/2.1.1.0/libexec/vendor/lib/python2.7/site-packages/OpenSSL/SSL.py", line 1232, in set_tlsext_host_name
    raise TypeError("name must be a byte string")
TypeError: name must be a byte string
```

It is related to https://github.com/pyca/pyopenssl/issues/15

Maybe updating pyOpenSSL to latest version (16.1.0) helps?

https://pypi.python.org/pypi/pyOpenSSL

https://github.com/Homebrew/homebrew-core/blob/f61e92830ed4d1828dd46f2a4fcce5202130351a/Formula/ansible.rb#L492

---

OS X 10.9.5
Python 2.7.9
Homebrew 0.9.9 (git revision 6622b; last commit 2016-09-01)
Homebrew/homebrew-core (git revision c2af; last commit 2016-09-01)

It works on another Computer with:

macOS 10.11.5
Python 2.7.10
also latest brew

**Comment by MikeMcQuaid** [comment]
> Maybe updating pyOpenSSL to latest version (16.1.0) helps?

Could you try and open a pull request? [This document](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) should help and I'm happy to walk you through anything else.

Thanks!

**Comment by tdsmith** [comment]
This sounds like an ansible-galaxy bug; it should be passing in byte strings, not Unicode strings.

**Comment by tdsmith** [comment]
This is either fixed by #4471 or not something we can fix on our end. :)

**Comment by davidak** [comment]
It is fixed with that update. Thanks!
